Lib/test/test_pyrepl/test_utils.py

```diff
@@ -60,3 +60,34 @@ def gen_raise():
         self.assertEqual(next(pnw), (3, 4, None))
         with self.assertRaises(ZeroDivisionError):
             next(pnw)
+
+    def test_gen_colors_keyword_highlighting(self):
+        no_highlight_cases = [
+            ("a.set", [(".", "op")]),
+            ("obj.list", [(".", "op")]),
+            ("obj.match", [(".", "op")]),
+        ]
+        for code, expected_highlights in no_highlight_cases:
+            with self.subTest(code=code):
+                colors = list(gen_colors(code))
+                # Extract (text, tag) pairs for comparison
+                actual_highlights = []
+                for color in colors:
+                    span_text = code[color.span.start:color.span.end + 1]
+                    actual_highlights.append((span_text, color.tag))
+                self.assertEqual(actual_highlights, expected_highlights,
+                               f"In '{code}', expected {expected_highlights}, got {actual_highlights}")
```

Comment:
```suggestion
                self.assertEqual(actual_highlights, expected_highlights)
```

The subtest() function will take care of indicating the issue.